lib/cadet/jobs/autograder/plagiarism_checker.ex

```diff
@@ -0,0 +1,70 @@
+defmodule Cadet.Autograder.PlagiarismChecker do
+  @moduledoc """
+  This module performs the plagiarism check script and sends results to
+  Amazon S3.
+  """
+  use Cadet, :context
+  use Que.Worker
+
+  require Logger
+
+  import Ecto.Query
+
+  @bucket_name :cadet
+               |> Application.fetch_env!(:plagiarism_check_vars)
+               |> Keyword.get(:bucket_name)
+
+  @plagiarism_script_path :cadet
+                          |> Application.fetch_env!(:plagiarism_check_vars)
+                          |> Keyword.get(:plagiarism_script_path)
+
+  def perform(assessment_id) when is_ecto_id(assessment_id) do
+    Logger.info("Running plagiarism check on Assessment #{assessment_id}")
+
+    {_, exit_result} =
+      System.cmd("python", [@plagiarism_script_path, "--assessment_id", to_string(assessment_id)])
+
+    exit_result
+    |> zip_results()
+    |> store()
+  end
+
+  defp store(assessment_id) when is_ecto_id(assessment_id) do
+    file_name = "assessment_#{assessment_id}.zip"
+
+    assessment_title =
+      Cadet.Assessments.Assessment
+      |> where(id: ^assessment_id)
+      |> select([a], a.title)
+      |> Repo.one()
+
+    response =
+      @bucket_name
+      |> ExAws.S3.put_object("/reports/assessment-#{assessment_title}", File.read!(file_name))
+      |> ExAws.request!()
+
+    # if response status < 400, the transaction was successful.
+    if Map.get(response, :status_code) < 400 do
+      File.rm(file_name)
+      File.rm_rf("submissions")
+    else
+      raise inspect(response)
+    end
+  end
+
+  defp zip_results(assessment_id) do
+    {_, zip_result} =
+      System.cmd("zip", [
+        "-r",
+        "assessment_#{assessment_id}.zip",
```

Comment:
You should probably put this instead the `submissions` directory so as to not dirty the root directory